powertools-idempotency/src/test/java/software/amazon/lambda/powertools/idempotency/persistence/cache/LRUCacheTest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package software.amazon.lambda.powertools.idempotency.persistence.cache;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class LRUCacheTest {
+
+    @Test
+    public void testLRUCache_shouldRemoveEldestEntry() {
+        LRUCache<String, String> cache = new LRUCache<>(3);
+        cache.put("key1", "value1");
+        cache.put("key2", "value2");
+        cache.put("key3", "value3");
+        cache.put("key4", "value4");
+        cache.put("key5", "value5");
+
+        assertThat(cache).hasSize(3);
+        assertThat(cache.get("key1")).isNull();
+        assertThat(cache.get("key2")).isNull();
```

Comment:
All 3 assert can be simplified to

```suggestion
        assertThat(cache)
                .hasSize(3)
                .doesNotContainKeys("key1", "key2");
```